core/src/main/java/com/datastrato/gravitino/storage/relational/service/RoleMetaService.java

```diff
@@ -120,14 +137,24 @@ public boolean deleteRole(NameIdentifier identifier) {
                 UserRoleRelMapper.class, mapper -> mapper.softDeleteUserRoleRelByRoleId(roleId)),
         () ->
             SessionUtils.doWithoutCommit(
-                GroupRoleRelMapper.class, mapper -> mapper.softDeleteGroupRoleRelByRoleId(roleId)));
+                GroupRoleRelMapper.class, mapper -> mapper.softDeleteGroupRoleRelByRoleId(roleId)),
+        () ->
+            SessionUtils.doWithoutCommit(
+                SecurableObjectMapper.class,
+                mapper -> mapper.softDeleteSecurableObjectsByRoleId(roleId)));
     return true;
   }
 
+  public List<SecurableObjectPO> listSecurableObjectsByRoleId(Long roleId) {
+    return SessionUtils.getWithoutCommit(
```

Comment:
I think some compatibility should be done when querying. First query the `role_meta_securable_object` table. If it does not exist, it will fall back to the `role_meta` table. Otherwise, the user will get wrong results when the data is not flushed.